label_studio/projects/functions/stream_history.py

```diff
@@ -0,0 +1,27 @@
+from django.db import transaction
+from django.conf import settings
+
+from projects.models import LabelStreamHistory
+
+
+def add_stream_history(next_task, user, project):
+    if next_task is not None:
+        with transaction.atomic():
+            history, created = LabelStreamHistory.objects.get_or_create(user=user, project=project)
+            new_history_data = {"taskId": next_task.id, "annotationId": None}
+            if created:
+                history.data = [new_history_data]
+            else:
+                task_ids = set([h['taskId'] for h in history.data])
+                if next_task.id not in task_ids:
+                    history.data.append(new_history_data)
+                if len(task_ids) + 1 > settings.LABEL_STREAM_HISTORY_LIMIT:
+                    history.data = history.data[-settings.LABEL_STREAM_HISTORY_LIMIT:]
+            history.save()
+
+
+def fill_history_annotation(user, task, annotation):
```

Comment:
do we need fill_history_draft? similar functionality, but for drafts

Reply:
@makseq can one annotator have several drafts for the same task? If not, so we don't need to select some particular one.

Reply:
if he is an admin, he can - via quickview

Reply:
@makseq Now user actions in quickview don't affect label stream history, if we want to change it, I think we need to collect product requirements first. I'm solving only the problem "history disappears after page refresh" described in the task.

Reply:
I guess the created drafts via quickview can create an issue when you move through history and don't specify which id LSF should use to load. E.g.:
1 user goes to label stream and postpone draft A
2 user creates draft B for task 1 in quickview
3 user goes back in task history and sees draft B instead of A